symphony/bdk/core/auth/obo_authenticator.py

```diff
@@ -3,26 +3,16 @@
 from abc import ABC, abstractmethod
 
 from symphony.bdk.core.auth.auth_session import OboAuthSession
-from symphony.bdk.core.auth.exception import AuthUnauthorizedError
 from symphony.bdk.core.auth.jwt_helper import create_signed_jwt
 from symphony.bdk.core.config.model.bdk_app_config import BdkAppConfig
-from symphony.bdk.gen.api_client import ApiClient
+from symphony.bdk.core.config.model.bdk_retry_config import BdkRetryConfig
+from symphony.bdk.core.retry import retry
+from symphony.bdk.core.retry.startegy import authentication_retry
 from symphony.bdk.gen.auth_api.certificate_authentication_api import CertificateAuthenticationApi
-from symphony.bdk.gen.exceptions import ApiException
 from symphony.bdk.gen.login_api.authentication_api import AuthenticationApi
 from symphony.bdk.gen.login_model.authenticate_request import AuthenticateRequest
 
 
-def wrap_api_exception(func):
-    async def try_and_wrap_api_exception(*args, **kwargs):
-        try:
-            return await func(*args, **kwargs)
-        except ApiException as exc:
-            raise AuthUnauthorizedError(OboAuthenticator.unauthorized_message) from exc
```

Comment:
Instead of raising an AuthUnauthorizedError, I am raising the ApiException (same retry strategy as the authentication)